web-client/src/pages/admin/AdminInventoryPage.tsx

```diff
@@ -1,5 +1,100 @@
-import UnderConstruction from "@/components/UnderConstruction";
+import { Button } from "@/components/ui/button";
+import { Search } from "lucide-react";
+import { ChevronDown } from "lucide-react";
+import { DropdownMenu, DropdownMenuItem, DropdownMenuTrigger, DropdownMenuContent } from "@/components/ui/dropdown-menu";
+import InventoryTable from '@/components/InventoryTable';
+import { useNavigate } from 'react-router-dom';
+import { useGetSocks } from '@/api/inventory/queries';
+import { useState } from 'react';
 
 export default function AdminInventoryPage() {
-  return <UnderConstruction pageName="Inventory" />;
+  const navigate = useNavigate();
+
+  const [page, setPage] = useState(1);
+  const limit = 8;
+  const offset = (page - 1) * limit;
+
+  const { data, error, isLoading } = useGetSocks(limit, offset);
+
+  if (isLoading) {
+    return <div>Loading...</div>;
+  }
+
+  if (error) {
+    return <div>Error fetching socks: {error instanceof Error ? error.message : 'Unknown error'}</div>;
+  }
+
+  if (!data || !data.items || data.items.length === 0) {
+    return <div>No socks available in the inventory.</div>;
+  }
+
+  const handleRowClick = (sockId: string) => {
+    navigate(`/socks/${sockId}`);
+  };
+
+  const totalSocks = data.total;
+  const totalPages = Math.ceil(totalSocks / limit);
+
+  const handleNextPage = () => {
+    if (page < totalPages) setPage((prev) => prev + 1);
+  };
+  const handlePreviousPage = () => {
+    if (page > 1) setPage((prev) => prev - 1);
+  };
+
+  const sortedSocks = data.items.sort((a, b) => a.id - b.id);
+
+  return (
+    <div className="admin-inventory-page">
+      <div className="flex justify-between items-center">
+        <h1 className="text-3xl font-extrabold">Inventory Management</h1>
+
+        <Button
+          className="ml-auto"
+          onClick={() => console.log('Navigate to Add New Product page')}
+        >
+          Add New Product
+        </Button>
+      </div>
+
+      <div className="search-filter flex items-center mt-4">
+        <div className="relative w-full">
+          <div className="absolute left-3 top-1/2 transform -translate-y-1/2">
+            <Search className="text-gray-500" size={20} />
+          </div>
+
+          <input
```

Comment:
Use the `Input` component from shadcn

Reply:
Updated:
<img width="765" alt="image" src="https://github.com/user-attachments/assets/10305d1f-1bee-474d-86bc-3f9d3d78842e">

Had to change the format a bit since the Search Icon was covering the text

Reply:
Yep sounds good! I guess it's a bit strange the Input from shadcn doesn't have "leading icon" slot. Like this other library: https://nextui.org/docs/components/input#start--end-content

If it gives you trouble you could probably just remove the icon really